pkg/cmd/validate.go

```diff
@@ -73,11 +97,12 @@ func NewRootCommand() *cobra.Command {
 		version:      "1.27",
 	}
 	res := &cobra.Command{
-		Use:   "kubectl-validate [manifests to validate]",
-		Short: "kubectl-validate",
-		Long:  "kubectl-validate is a CLI tool to validate Kubernetes manifests against their schemas",
-		Args:  cobra.MinimumNArgs(1),
-		RunE:  invoked.Run,
+		Use:          "kubectl-validate [manifests to validate]",
+		Short:        "kubectl-validate",
+		Long:         "kubectl-validate is a CLI tool to validate Kubernetes manifests against their schemas",
+		Args:         cobra.MinimumNArgs(1),
+		RunE:         invoked.Run,
+		SilenceUsage: true,
```

Comment:
When a command returns an error, cobra will automatically print the usage text. It means that Cobra assumes if the command returns an error, it's because of a mistake in usage, and we need to show the usage message to the user. But in our case (returning an error because of invalid manifests) we don't need to print usage.

P.S. There were some discussions about it in [here](https://github.com/spf13/cobra/issues/340).

However, I need to know your opinion. I also thought that we can use `os.Exit` at the end of the command's `Run` function, but also thought it wouldn't be a standard way to return exit codes in the commands. So, the only way I could think of was to silence usage and handle the exit code in the main function. The user still can print usage with `--help`.

Reply:
Yeah, ok no that sounds good. We definitely don't want to print usage, especially since the usage is super simple here.